static/js/popover_menus.js

```diff
@@ -85,4 +150,80 @@ export function initialize() {
             compose_mobile_button_popover_displayed = false;
         },
     });
+
+    function schedule_message_to_custom_date(send_at_time) {
+        if (!send_at_time) {
+            return;
+        }
+
+        const request = compose.create_message_object();
+        request.content = `/schedule ${send_at_time}\n` + request.content;
+        reminder.schedule_message(request);
+    }
+
+    function set_compose_box_schedule(element) {
+        const send_later_in = $(element).attr("id");
+        if ($(element).attr("class") === "send_later_hours") {
+            const send_at_time = format(
+                add(new Date(), {minutes: send_later_hours[send_later_in].hours * 60}),
+                "MMM d yyyy h:mm a",
+            );
+            return send_at_time;
+        } else if ($(element).attr("class") === "send_later_tomorrow") {
+            const send_time = send_later_tomorrow[send_later_in].time;
+            const date = new Date();
+            const scheduled_date = date.setDate(date.getDate() + 1);
+            const send_at_time = format(scheduled_date, "MMM d yyyy ") + send_time;
+            return send_at_time;
+        } else if ($(element).attr("class") === "send_later_days_and_weeks") {
+            const send_at_time = format(
+                add(new Date(), {days: send_later_days_and_weeks[send_later_in].days}),
+                "MMM d yyyy h:mm a",
+            );
+            return send_at_time;
+        }
+        blueslip.error("Not a valid time.");
+        return false;
+    }
+
+    delegate("body", {
+        ...default_popover_props,
+        target: "#send_later i",
+        onShow(instance) {
+            popovers.hide_all_except_sidebars(instance);
+            instance.setContent(
+                render_send_later_popover({
+                    send_later_hours,
+                    send_later_tomorrow,
+                    send_later_days_and_weeks,
+                    send_later_custom,
+                }),
+            );
+
+            compose_send_later_popover_displayed = true;
+            $(instance.popper).one("click", instance.hide);
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            $popper.one("click", "#send-later-custom-input", () => {
+                show_flatpickr($("#send_later")[0], schedule_message_to_custom_date, new Date());
+            });
+
+            $popper.one(
+                "click",
+                ".send_later_hours, .send_later_tomorrow, .send_later_days_and_weeks",
+                (e) => {
+                    const send_at_time = set_compose_box_schedule(e.currentTarget);
+                    schedule_message_to_custom_date(send_at_time);
+
+                    // We need the click event to propagate so that we can hide the popover.
```

Comment:
I guess the alternative would be to have `stopPropagation` and explicit hiding of the popover here?  That approach might be more readable.

Reply:
Okay